Mathlib/RingTheory/TwoSidedIdeal/Operations.lean

```diff
@@ -0,0 +1,207 @@
+/-
+Copyright (c) 2024 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import Mathlib.RingTheory.TwoSidedIdeal.Lattice
+import Mathlib.RingTheory.TwoSidedIdeal.BigOperators
+import Mathlib.RingTheory.Congruence.Opposite
+import Mathlib.Algebra.BigOperators.Ring
+import Mathlib.Data.Fintype.BigOperators
+import Mathlib.RingTheory.Ideal.Basic
+import Mathlib.Order.GaloisConnection
+
+
+/-!
+# Operations on Two-Sided-Ideals
+
+This file defines operations on two-sided-ideals of a ring `R`.
+
+## Main definitions and results
+- `TwoSidedIdeal.span`: the span of `s ⊆ R` is the smallest two-sided-ideal containing the set.
+- `TwoSidedIdeal.mem_span_iff_exists`: an element `x` is in the span of `s` if and only if it can be
```

Comment:
```suggestion
- `TwoSidedIdeal.mem_span_iff_exists`: in a unital and associative ring, an element `x` is in the span of `s` if and only if it can be
```